src/ZfrCors/Factory/CorsServiceFactory.php

```diff
@@ -35,11 +36,20 @@ class CorsServiceFactory implements FactoryInterface
      * {@inheritDoc}
      * @return CorsService
      */
-    public function createService(ServiceLocatorInterface $serviceLocator)
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
     {
         /* @var $corsOptions CorsOptions */
-        $corsOptions = $serviceLocator->get('ZfrCors\Options\CorsOptions');
+        $corsOptions = $container->get('ZfrCors\Options\CorsOptions');
 
         return new CorsService($corsOptions);
     }
+    /**
+     * {@inheritDoc}
+     *
+     * @return CorsService
+     */
+    public function createService(ServiceLocatorInterface $container)
```

Comment:
@svycka Can you point to the test failures? I could look at them to give guidance.

What @macnibblet is suggesting is the route we went with factories in Apigility itself: they do not implement `FactoryInterface`, but instead are simply callable; additionally, we don't typehint the container. This provides forwards compatibility out-of-the-box. As such, I'd like to see the test failures to understand what exactly is failing; is it tests of the factories themselves? or some integration test?

(Side note: The only place we'll likely have issues in Apigility is where we define abstract factories, and those will need to be updated.)